lib/highline/question/answer_converter.rb

```diff
@@ -0,0 +1,84 @@
+# coding: utf-8
+
+require 'forwardable'
+
+class HighLine
+  class Question
+    class AnswerConverter
+      extend Forwardable
+
+      def_delegators :@question,
+                     :answer, :answer=, :check_range,
+                     :directory, :answer_type, :choices_complete
+
+      def initialize(question)
+        @question = question
+      end
+
+      def convert
+        return unless answer_type
+
+        self.answer = convert_by_answer_type
+        check_range
+        answer
+      end
+
+      private
+
+      def convert_by_answer_type
+        if answer_type.respond_to? :parse
+          answer_type.parse(answer)
+        elsif answer_type.is_a? Class
+          send(answer_type.name)
+        else
+          send(answer_type.class.name)
+        end
+      end
+
+      def String
+        HighLine::String(answer)
+      end
+
+      # That's a weird name for a method!
+      # But it's working ;-)
+      define_method "HighLine::String" do
+        HighLine::String(answer)
+      end
+
+      def Integer
+        Kernel.send(:Integer, answer)
```

Comment:
What you think about [#public_send](http://ruby-doc.org/core-2.1.1/Object.html#method-i-public_send) over #send